pkgs/by-name/st/stdoutisatty/package.nix

```diff
@@ -0,0 +1,70 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  makeBinaryWrapper,
+  nix-update-script,
+  runCommand,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "stdoutisatty";
+  version = "1.0";
+
+  src = fetchFromGitHub {
+    owner = "lilydjwg";
+    repo = finalAttrs.pname;
+    rev = finalAttrs.version;
+    hash = "sha256-NyVn9cxx0rY1ridNDTqe0pzcVhdLVaPCKT4hoQkQzRs=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    makeBinaryWrapper
+  ];
+
+  preFixup = ''
+    # necessary for dynamic loading of `lib/libstdoutisatty.so` at runtime
+    wrapProgram $out/bin/${finalAttrs.pname} \
+      --prefix LD_LIBRARY_PATH : "$out/lib"
+  '';
+
+  passthru = {
+    updateScript = nix-update-script { };
+    tests = {
+      ls-color = runCommand "${finalAttrs.pname}-ls-color" { } ''
```

Comment:
Or we can use `tty -s` to simplify the test: https://stackoverflow.com/a/10022365

Reply:
This is interesting! Although I couldn't actually get `tty` to reliably determine whether `stdout` is a tty... For example, `faketty tty <&1 | less` prints `not a tty`, so I don't know how to proceed at the moment.